### ISSUE TITLE
What's the recommended way to wait for Dapr K8s deployment from Helm Chart?

### ISSUE THREAD

**Issue by seanyen** [dapr/quickstarts#402]
## Ask your question here

We are using the Helm to deploy Dapr in a cluster in a CI/CD pipeline and this is the used command:

```bash
sudo helm install dapr dapr/dapr --create-namespace --namespace dapr-system --version 1.0.1 --debug --wait
```

However, many times it is just failing at talking to dashbaord service. (Sometimes it could go through, but the chance is low.)

Not sure if I am looking at the wrong approach, or am I hitting an issue.

```
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /home/vsts/work/_temp/0b9668da-c2ef-4761-b650-cb3e797ab9d9.sh
install.go:173: [debug] Original chart version: "1.0.1"
install.go:190: [debug] CHART PATH: /home/vsts/.cache/helm/repository/dapr-1.0.1.tgz

client.go:122: [debug] creating 1 resource(s)
client.go:122: [debug] creating 1 resource(s)
client.go:122: [debug] creating 1 resource(s)
install.go:156: [debug] Clearing discovery cache
wait.go:53: [debug] beginning wait for 3 resources with timeout of 1m0s
client.go:122: [debug] creating 1 resource(s)
client.go:122: [debug] creating 23 resource(s)
wait.go:53: [debug] beginning wait for 23 resources with timeout of 5m0s
wait.go:244: [debug] Deployment is not ready: dapr-system/dapr-dashboard. 0 out of 1 expected pods are ready
wait.go:244: [debug] Deployment is not ready: dapr-system/dapr-dashboard. 0 out of 1 expected pods are ready
Error: Get "https://10.1.0.4:16443/api/v1/namespaces/dapr-system/services/dapr-dashboard": dial tcp 10.1.0.4:16443: connect: connection refused
helm.go:81: [debug] Get "https://10.1.0.4:16443/api/v1/namespaces/dapr-system/services/dapr-dashboard": dial tcp 10.1.0.4:16443: connect: connection refused
```

**Comment by yoDon** [comment]
See also https://github.com/dapr/dapr/issues/2861 (which asks for a --wait flag on the dapr command) and https://github.com/dapr/quickstarts/issues/398 (which as of this writing is marked as Closed but which has not actually been resolved)

**Comment by yoDon** [comment]
If the issue is just the dashboard service that may be the time required for the dashboard web server to startup inside the container (helm wait only knows how to wait until the pod is ready, not until the process running inside the pod is ready). Adding a couple seconds of sleep wait after the helm wait should give the dashboard process time to start inside the container.

**Comment by wcs1only** [comment]
Correct, this. indicates that the dashboard service is unavailable. We'd need to understand why the dashboard pod hasn't come up to diagnose further.

Possible reasons might be that the cluster is resource constrained. These commands might tell you more:
```
kubectl describe pod -l app=dapr-dashboard -n dapr-system
kubectl logs -l app=dapr-dashboard -n dapr-system
```

Also, for support requests, the fastest way to get an answer is to head down to our community discord: 
https://aka.ms/dapr-discord

**Comment by wcs1only** [comment]
This issue hasn't had any activity for 30+ days. Please open a new issue if you are still having trouble, or head over to our community discord https://aka.ms/dapr-discord.
